tests/python/gaia-ui-tests/gaiatest/apps/contacts/app.py

```diff
@@ -149,7 +149,7 @@ def _return_class_from_tap(self, return_class='ContactDetails'):
                 return ContactDetails(self.marionette)
             elif return_class == 'EditContact':
                 # This may seem superfluous but we can enter EditContact from Contacts, or from ActivityPicker
-                Wait(self.marionette).until(lambda m: self.apps.displayed_app.name == Contacts.name)
+                Wait(self.marionette).until(lambda m: self.apps.displayed_app.manifest_url == Contacts.manifest_url)
```

Comment:
You can change this for Contacts(self.marionette).wait_to_be_displayed() , I think